File: User-Preferences-Service/weather-aggregator-user-preferences-service-app/src/org/unibl/etf/model/requests/update_user_preferences_request.rs

use serde::Deserialize;
use uuid::Uuid;
use crate::org::unibl::etf::model::domain::entities::user_preferences_entity::unit_system_type::UnitSystemType;


#[derive(Debug, Deserialize)]
pub struct UpdateUserPreferencesRequest {
    pub user_id: Uuid,
    pub unit_system: UnitSystemType,
    pub favorite_location_name: Option<String>,
    pub favorite_lat: Option<f64>,
    pub favorite_lon: Option<f64>,
}

File: web-frontend/weather-aggregator-frontend/src/model/requests/UpdateUserPreferencesRequest.ts
import type { UnitSystemType } from "../UserPreferencesWithLocationHistory";

export interface UpdateUserPreferencesRequest {
  user_id: string; // Uuid is handled as a string
  unit_system: UnitSystemType; // Assuming UnitSystemType is "metric" | "imperial"
  favorite_location_name?: string | null;
  favorite_lat?: number | null;
  favorite_lon?: number | null;
}

